test/tstQueryTreeManufacturedSolution.cpp

```diff
@@ -38,16 +38,22 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(structured_grid, TreeTypeTraits,
   using Point = ArborX::Point<DIM, Coordinate>;
   using Box = ArborX::Box<DIM, Coordinate>;
 
-  // FIXME_NVCC we see inexplainable test failures with NVCC and KDOP<18> and
-  // KDOP<26> here.
-#ifdef __NVCC__
   if constexpr (ArborX::GeometryTraits::is_kdop_v<BoundingVolume>)
   {
+#ifdef __NVCC__
+    // FIXME_NVCC we see inexplainable test failures with NVCC and KDOP<18> and
+    // KDOP<26> here.
     if constexpr (BoundingVolume::n_directions == 9 ||
                   BoundingVolume::n_directions == 13)
       return;
-  }
 #endif
+#ifdef __INTEL_LLVM_COMPILER
```

Comment:
Does it also fail when using `icpx` with host backends? Otherwise, I would rather check for `KOKKOS_ENABLE_SYCL`.

Reply:
I'll need to check that.

Reply:
It seems pretty intermittent in failures in our CI. Can't reproduce it on Aurora.

Reply:
Use `KOKKOS_IF_ON_HOST/DEVICE` which uses `__SYCL_DEVICE_ONLY__`.

Reply:
This isn't actually in device code, so you should use `if constexpr(std::is_same_v<ExecutionSpace, Kokkos::SYCL>)` instead.